lib/LanguageServerRename/Handler/RenameHandler.php

```diff
@@ -99,11 +101,12 @@ public function registerCapabiltiies(ServerCapabilities $capabilities): void
         $capabilities->renameProvider = new RenameOptions(true);
     }
     
-    private function resultToWorkspaceEdit(iterable $results): WorkspaceEdit
+    private function resultToWorkspaceEdit(Traversable $results): WorkspaceEdit
     {
         $documentEdits = [];
-        foreach ($results as $result) {
-            /** @var RenameResult $result */
+        $map = LocatedTextEditsMap::fromLocatedEdits(iterator_to_array($results));
```

Comment:
this will be blocking, use `foreach ($results as $result)` and potentially adda `delay(1)` every so often to unblock.

Reply:
Currently I haven't had a slow rename, so not sure it will be necessary.

Reply:
Try looking for a method which is like `next` in a big project, it will probably time out after 10 seconds.